src/DotVVM.Framework/Routing/RouteHelper.cs

```diff
@@ -35,5 +36,54 @@ public static void AutoDiscoverRoutes(this DotvvmRouteTable table, IRoutingStrat
                 table.Add(route.RouteName, route);
             }
         }
+
+        public static void AssertConfigurationIsValid(this DotvvmConfiguration config)
+        {
+            var invalidRoutes = new List<DotvvmConfigurationAssertResult<RouteBase>>();
+            var invalidControls = new List<DotvvmConfigurationAssertResult<DotvvmControlConfiguration>>();
+            var loader = new AggregateMarkupFileLoader();
+            foreach (var route in config.RouteTable.Where(s => !string.IsNullOrWhiteSpace(s.VirtualPath)))
+            {
+                if (string.IsNullOrWhiteSpace(route.RouteName))
+                {
+                    invalidRoutes.Add(new DotvvmConfigurationAssertResult<RouteBase>(route, DotvvmConfigurationAssertReason.MissingRouteName));
+                }
+
+                var content = loader.GetMarkup(config, route.VirtualPath);
+                if (content == null)
+                {
+                    invalidRoutes.Add(new DotvvmConfigurationAssertResult<RouteBase>(route, DotvvmConfigurationAssertReason.MissingFile));
+                }
+            }
+            foreach (var control in config.Markup.Controls.Where(s => !string.IsNullOrWhiteSpace(s.Src)))
+            {
+                if (string.IsNullOrWhiteSpace(control.TagPrefix))
+                {
+                    invalidControls.Add(new DotvvmConfigurationAssertResult<DotvvmControlConfiguration>(control, DotvvmConfigurationAssertReason.MissingControlTagPrefix));
+                    break;
```

Comment:
mmm, what about `else`?

Reply:
@exyi `else`? what about other controls without `TagPrefix`

Reply:
Sorry, I thought this was a `continue` statement... Anyway, to be explicit - the `break` statement jumps after the `foreach` statement which means that only one error is added to the collection `invalidControls`. Why is it a collection then? I'd change it to a series of `if ... else` statements and add exactly one or zero errors **per registered control**